Add dormitory option to group mail feature

mabezi opened the pull request:
The title says it all. Dormitories should remain optional so it's still possible to send a mail to all members.

An idea for the future would be to migrate dormitories and groups into multiple-choice fields.

mabezi committed:
Add dormitory option to group mail feature

agmes4 commented on pycroft/lib/user/mail.py:
```diff
@@ -125,14 +126,24 @@ def user_send_mail(
+def group_send_mail(
+    group: PropertyGroup, subject: str, body_plain: str, building: Building | None = None
+) -> None:
+    users = get_active_users(session=session.session, group=group, building=building)
```
```suggestion
    users = get_active_users(session=session.session, group=group, buildings=buildings)
```

agmes4 commented on web/blueprints/user/__init__.py:
```diff
@@ -1543,6 +1543,7 @@ def mail_group() -> ResponseReturnValue:
             group=form.group.data,
             subject=form.subject.data,
             body_plain=form.body_plain.data,
+            building=form.building.data if getattr(form, "building", None) else None,
```
```suggestion
            buildings=form.buildings.data if getattr(form, "buildings", None) else [],
```

agmes4 committed:
Add test for mail feature

agmes4 committed:
Updates send mail function

Updates send email function in order to select multiple groups as well
as buildings
